edx_proctoring/static/index.js

```diff
@@ -0,0 +1,32 @@
+export const handlerWrapper = (Handler) => {
+   let handler = new Handler({});
+
+  self.addEventListener("message", (message) => {
+    switch(message.data.type) {
+      case 'config': {
+        handler = new Handler(message.data.options);
+        break;
+      }
+      case 'startExamAttempt': {
+        if(handler.onStartExamAttempt) {
+          handler.onStartExamAttempt().then(() => self.postMessage({type: 'examAttemptStarted'}))
```

Comment:
I would pull these type strings out into constants so they can be imported for use here and in the action handler

Reply:
I'm also not clear on how this interfaces with the proctoring tool but you may want to accept some data object that can be passed to the action handler. For example would they need some identifier of the exam being taken? Again this is my first time looking at this and I don't think I have the full picture so please ignore if this doesn't make any sense.